django/middleware/locale.py

```diff
@@ -16,28 +16,35 @@ class LocaleMiddleware(MiddlewareMixin):
 
     response_redirect_class = HttpResponseRedirect
 
+    def get_fallback_language(self, request):
+        # Allows for easier fallback logic by extending LocaleMiddleware
+        # and overriding this method. Fallback to supported languages
+        # if a variant is used
+        try:
+            return translation.get_supported_language_variant(settings.LANGUAGE_CODE)
+        except LookupError:
+            return settings.LANGUAGE_CODE
+
     def process_request(self, request):
         urlconf = getattr(request, "urlconf", settings.ROOT_URLCONF)
         (
             i18n_patterns_used,
             prefixed_default_language,
         ) = is_language_prefix_patterns_used(urlconf)
+
```

Comment:
Did Black add this? 🤔  (Usually better to leave out miscellaneous whitespace changes if not)

Reply:
Oh, no, this was an accident, although for me it greatly improves readability :) 
Fixed, Thanks!